colorbleed/scripts/fusion_switch_shot.py

```diff
@@ -0,0 +1,213 @@
+import os
+import sys
+import logging
+
+# Pipeline imports
+from avalon import api, io, pipeline
+import avalon.fusion
+
+# Config imports
+import colorbleed.lib as colorbleed
+import colorbleed.fusion.lib as fusion_lib
+
+log = logging.getLogger("Update Slap Comp")
+
+self = sys.modules[__name__]
+self._project = None
+fusion = None
+
+
+def _get_work_folder(session):
+    """Convenience function to get the work folder path of the current asset"""
+
+    # Get new filename, create path based on asset and work template
+    template_work = self._project["config"]["template"]["work"]
+    work_path = pipeline._format_work_template(template_work, session)
+
+    return os.path.normpath(work_path)
+
+
+def _get_fusion_instance():
+    fusion = getattr(sys.modules["__main__"], "fusion", None)
+    if fusion is None:
+        try:
+            # Support for FuScript.exe, BlackmagicFusion module for py2 only
+            import BlackmagicFusion as bmf
+            fusion = bmf.scriptapp("Fusion")
+        except ImportError:
+            raise RuntimeError("Could not find a Fusion instance")
+    return fusion
+
+
+def _format_filepath(session):
+
+    project = session["AVALON_PROJECT"]
+    asset = session["AVALON_ASSET"]
+
+    # Save updated slap comp
+    work_path = _get_work_folder(session)
+    walk_to_dir = os.path.join(work_path, "scenes", "slapcomp")
+    slapcomp_dir = os.path.abspath(walk_to_dir)
+
+    # Ensure destination exists
+    if not os.path.isdir(slapcomp_dir):
+        log.warning("Folder did not exist, creating folder structure")
+        os.makedirs(slapcomp_dir)
+
+    # Compute output path
+    new_filename = "{}_{}_slapcomp_v001.comp".format(project, asset)
+    new_filepath = os.path.join(slapcomp_dir, new_filename)
+
+    # Create new unqiue filepath
+    if os.path.exists(new_filepath):
+        new_filepath = colorbleed.version_up(new_filepath)
+
+    return new_filepath
+
+
+def _update_savers(comp, session):
+    """Update all savers of the current comp to ensure the output is correct
+
+    Args:
+        comp (object): current comp instance
+        asset (dict): asset document of the asset to update TO
+
+    Returns:
+         None
+    """
+
+    new_work = _get_work_folder(session)
+    # TODO
```

Comment:
This doesn't seem finished. :) (also see the line below it with the question mark, hehe)